### PR TITLE
Require Go 1.4 for OpenShift

### PR DESCRIPTION
**Pull request by smarterclayton**
Cannot be merged until centos7 has Go 1.4 available as RPMs.

### CODE REVIEW

**Comment by smarterclayton** [comment]
Covers https://github.com/openshift/origin/pull/1070

File: hack/common.sh [hunk 1]
```diff
@@ -197,12 +197,12 @@ EOF
...
     local go_version
     go_version=($(go version))
-    if [[ "${go_version[2]}" < "go1.2" ]]; then
+    if [[ "${go_version[2]}" < "go1.4" ]]; then
       echo <<EOF
 
 Detected go version: ${go_version[*]}.
-Kubernetes requires go version 1.2 or greater.
-Please install Go version 1.2 or later.
+OpenShift and Kubernetes requires go version 1.4 or greater.
```
Comment (ncdc) [inline hack/common.sh]: s/requires/require/

File: hack/verify-gofmt.sh [hunk 2]
```diff
@@ -10,7 +10,7 @@ GO_VERSION=($(go version))
 
 GO_VERSION=($(go version))
 
-if [[ -z $(echo "${GO_VERSION[2]}" | grep -E 'go1.2|go1.3') ]]; then
+if [[ -z $(echo "${GO_VERSION[2]}" | grep -E 'go1.2|go1.3|go1.4') ]]; then
```
Comment (ncdc) [inline hack/verify-gofmt.sh]: If we're going to require 1.4, why keep 1.2 & 1.3 here?

File: hack/verify-golint.sh [hunk 3]
```diff
@@ -11,7 +11,7 @@ fi
 
 GO_VERSION=($(go version))
 
-if [[ -z $(echo "${GO_VERSION[2]}" | grep -E 'go1.2|go1.3') ]]; then
+if [[ -z $(echo "${GO_VERSION[2]}" | grep -E 'go1.2|go1.3|go1.4') ]]; then
```
Comment (ncdc) [inline hack/verify-golint.sh]: If we're going to require 1.4, why keep 1.2 & 1.3 here?

File: hack/verify-gofmt.sh (hunk 2 above)
Comment (smarterclayton) [inline hack/verify-gofmt.sh]: Because

Comment (ncdc) [inline hack/verify-gofmt.sh]: Of course

**Comment by sdodson** [comment]
Mind bumping the rpm build requires in the same PR?
https://github.com/openshift/origin/blob/master/openshift.spec#L28

**Comment by sdodson** [comment]
Should be fine to merge this WRT to RHEL builds. I don't know for certain that golang-1.4 is available for Centos yet.

**Comment by smarterclayton** [comment]
I don't believe it is, yet.

**Comment by mfojtik** [comment]
@smarterclayton it is.

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**